### ISSUE TITLE
The shuffle=True argument in SFT is not used

### ISSUE THREAD

**Issue by shengyangs** [NVIDIA/NeMo-Aligner#29]
**Describe the bug**

The SFT trainer has no data shuffling mechanism even when the `shuffle=True` in config.

**Steps/Code to reproduce bug**

Below is the data-related config I used. 

```
data:
        chat: true
        chat_prompt_tokens:
          system_turn_start: "\0"
          turn_start: "\x11"
          label_start: "\x12"
          end_of_turn: '
    
            '
          end_of_name: '
    
            '
        sample: false
        num_workers: 1
        dataloader_type: single
        train_ds:
          file_path: /home/shengyangs/data/ultrachat_200k_nmt/ultrachat_200k_sft_all.dialog.jsonl
          global_batch_size: 1
          micro_batch_size: 1
          shuffle: true
          memmap_workers: null
          max_seq_length: 4096
          min_seq_length: 1
          drop_last: true
          concat_sampling_probabilities: null
          label_key: output
          add_eos: false
          add_sep: false
          add_bos: false
          truncation_field: input
          index_mapping_dir: /home/shengyangs/data/indexmap_dir
          prompt_template: '{input} {output}'
          hf_dataset: true
          truncation_method: right
        validation_ds:
          file_path: /home/shengyangs/data/ultrachat_200k/ultrachat_200k_test_sft.jsonl
          names: null
          global_batch_size: 1
          micro_batch_size: 1
          shuffle: false
          memmap_workers: ${model.data.train_ds.memmap_workers}
          max_seq_length: 4096
          min_seq_length: 1
          drop_last: false
          label_key: ${model.data.train_ds.label_key}
          add_eos: false
          add_sep: ${model.data.train_ds.add_sep}
          add_bos: ${model.data.train_ds.add_bos}
          write_predictions_to_file: false
          output_file_path_prefix: null
          truncation_field: ${model.data.train_ds.truncation_field}
          index_mapping_dir: /home/shengyangs/data/indexmap_dir
          prompt_template: ${model.data.train_ds.prompt_template}
          tokens_to_generate: 32
          hf_dataset: true
          truncation_method: right
          metric:
            name: loss
            average: null
            num_classes: null
        test_ds:
          prompt_template: ${model.data.train_ds.prompt_template}
```

With this config, the first processed training example is still the first row of my input dataset.

Looking into the calling stack, the `shuffle` argument is not used when building the dataset or dataloader.

**Expected behavior**

The training data should be shuffled when `data.train_ds.shuffle=True`
